compiler/rustc_traits/src/normalize_erasing_regions.rs

```diff
@@ -15,10 +15,19 @@ crate fn provide(p: &mut Providers) {
                 .perf_stats
                 .normalize_generic_arg_after_erasing_regions
                 .fetch_add(1, Ordering::Relaxed);
-            normalize_after_erasing_regions(tcx, goal)
+
+            let (param_env, goal) = goal.into_parts();
+            tcx.try_normalize_erasing_regions(param_env, goal).unwrap_or_else(|_| bug!(
```

Comment:
Wasn't sure whether it's better to use `try_normalize_generic_arg_after_erasing_regions` here directly or implement this as here (and go through the visitor). The latter would lead to more caching, which might be beneficial when calling `normalize_erasing_regions` in later stages of compilation, or we might just cache stuff we don't need and also have the cost of traversing the visitor on each call, not sure.